integration-libs/opf/base/root/components/opf-payment-verification/opf-payment-verification.component.ts

```diff
@@ -30,19 +34,45 @@ export class OpfPaymentVerificationComponent implements OnInit, OnDestroy {
     this.subscription = this.paymentService
       .verifyResultUrl(this.route)
       .pipe(
-        concatMap(({ paymentSessionId, responseMap }) => {
-          return this.paymentService.verifyPayment(
+        concatMap(
+          ({
             paymentSessionId,
-            responseMap
-          );
-        }),
-        concatMap(() => {
-          return this.paymentService.placeOrder();
-        })
+            paramsMap: paramsMap,
+            afterRedirectScriptFlag,
+          }) => {
+            if (afterRedirectScriptFlag === 'true') {
```

Comment:
I think it also will be good to move to separate method, which will return proper method from payment service based on condition

Reply:
thx!divided in more comprehensive methods:
runPaymentPattern
  runHostedFieldsPattern
  runHostedPagePattern